sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -232,7 +234,7 @@ final class DataFrameWriter private[sql](df: DataFrame) {
    * @since 1.4.0
    */
   @scala.annotation.varargs
-  def partitionBy(colNames: String*): DataFrameWriter = {
+  def partitionBy(colNames: String*): this.type = {
```

Comment:
Forgot to change it. No difference in byte codes but it will show `DataFrameWriter.this.type` in scala doc. I will update it.